mono/utils/mono-threads-state-machine.c

```diff
@@ -322,6 +396,13 @@ mono_threads_transition_request_resume (MonoThreadInfo* info)
 
 			return ResumeOk; //Resume worked and there's nothing for the caller to do.
 		} else {
+			if (cur_state == STATE_BLOCKING_ASYNC_SUSPENDED) {
```

Comment:
We should probably just have it in a different case statement to ease undestanding the code and keep it consistent with the rest